mono/sgen/sgen-protocol.c

```diff
@@ -100,12 +116,19 @@ void
 binary_protocol_init (const char *filename, long long limit)
 {
 #ifdef HAVE_UNISTD_H
-	filename_or_prefix = (char *)sgen_alloc_internal_dynamic (strlen (filename) + 1, INTERNAL_MEM_BINARY_PROTOCOL, TRUE);
-	strcpy (filename_or_prefix, filename);
-
 	file_size_limit = limit;
 
-	binary_protocol_open_file ();
+	/* Original name length + . + pid length in hex + null terminator */
+	filename_or_prefix = (char *)sgen_alloc_internal_dynamic (strlen (filename) + 1 + 8 + 1, INTERNAL_MEM_BINARY_PROTOCOL, TRUE);
```

Comment:
Also this length calculation makes me nervous.  Please use the `snprintf` technique:

    http://stackoverflow.com/questions/8090888/what-is-the-max-possible-length-of-a-pid-of-a-process-64-bit